src/cart/crt_register.c

```diff
@@ -629,9 +629,13 @@ crt_proto_query_int(crt_endpoint_t *tgt_ep, crt_opcode_t base_opc, uint32_t *ver
 	proto_query->pq_user_arg = arg;
 	proto_query->pq_coq->coq_base = base_opc;
 
-	rc = crt_req_set_timeout(rpc_req, timeout);
-	/** Should only fail if invalid parameter */
-	D_ASSERT(rc == 0);
+	if (timeout != 0) {
+		if (timeout < crt_gdata.cg_timeout) {
```

Comment:
i would remove this line, as crt_gdata.cg_timeout isnt necessarily the one being used either if per-rpc timeout isnt set. the order is -- global timeout (crt_gdata.cg_timeout), per-context timeout, per-rpc timeout.

Reply:
So we should use the crt_req_get_timeout() instead to get the default?